### PR TITLE
[SGDv2] Tensorboard Callback

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

This PR will add a Tensorboard logging callback to SGDv2. It also modifies the existing callback API to pass the logdir from the Trainer.

Depends on https://github.com/ray-project/ray/pull/17807

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/sgd/v2/callbacks/logging.py
Comment (amogkam) [inline python/ray/util/sgd/v2/callbacks/logging.py:215]: Probably out of scope for this PR, but is this logic something that can be abstracted away into `ml_utils` and then just called by both the Tune logger and SGD logger? Same for the Json logger

**Review by amogkam — COMMENTED** [review]
Nice, LGTM! I'll do a final pass and we can merge this in after #17807 gets merged.

File: python/ray/util/sgd/v2/callbacks/logging.py
Comment (Yard1) [reply python/ray/util/sgd/v2/callbacks/logging.py:215]: Yeah, I was thinking about that. It could be made generic. I'll make an issue and later, a follow-up PR.

**Comment by amogkam** [comment]
@Yard1 can you merge in master now that https://github.com/ray-project/ray/pull/17807 is merged. Thanks!

**Comment by Yard1** [comment]
@amogkam Merged!

File: python/ray/util/sgd/v2/callbacks/logging.py
Comment (amogkam) [inline python/ray/util/sgd/v2/callbacks/logging.py:145]: Would `worker_to_log` be a better name here instead of `workers_to_log`, since it's only 1 worker?

File: python/ray/util/sgd/v2/callbacks/logging.py
Comment (amogkam) [inline python/ray/util/sgd/v2/callbacks/logging.py:185]: If we're going to ignore `filename` argument then do we need to even accept it as an arg?

**Review by amogkam — APPROVED** [review]
Nice! Just left some minor comments.

File: python/ray/util/sgd/v2/callbacks/logging.py
Comment (Yard1) [reply python/ray/util/sgd/v2/callbacks/logging.py:145]: The argument name is left as-is to maintain compatibility with the parent class.

File: python/ray/util/sgd/v2/callbacks/logging.py
Comment (Yard1) [reply python/ray/util/sgd/v2/callbacks/logging.py:185]: As above

File: python/ray/util/sgd/v2/callbacks/logging.py
Comment (amogkam) [reply python/ray/util/sgd/v2/callbacks/logging.py:145]: Isn't it fine though if this class has different args than the parent class?

Comment (Yard1) [reply python/ray/util/sgd/v2/callbacks/logging.py:145]: It wouldn't break anything, though it seems off to me from an OOP design standpoint. It's my belief that every subclass should have the same arguments and attributes as the parent class, plus any extra ones - but it should not remove or modify the args/attrs of the parent. That being said `worker_to_log` would be more in line with what's actually happening. Let me know what you think, I'm fine with either.

**Review by amogkam — CHANGES_REQUESTED** [review]
See comment

**Review by matthewdeng — COMMENTED** [review]
Thanks for doing this! Would you able to run an example with this new callback and include the results in the PR description? Something we can follow along for testing/reproducibility/porting over to the docs later!

**Review by amogkam — COMMENTED** [review]
Once again, thanks a lot for doing this and laying the groundwork for all the Callbacks. I left a couple more comments about if we can make the logging mixin more generic and move the single file logic to the single file callback.

**Review by amogkam — APPROVED** [review]
Looks great, thanks!

**Pull request merged**
